kafka-avro/src/main/kotlin/com/gxf/utilities/kafka/avro/AvroSerializer.kt

```diff
@@ -6,24 +6,37 @@ SPDX-License-Identifier: Apache-2.0
 package com.gxf.utilities.kafka.avro
 
 import org.apache.avro.message.BinaryMessageEncoder
-import org.apache.avro.specific.SpecificRecord
+import org.apache.avro.specific.SpecificData
+import org.apache.avro.specific.SpecificRecordBase
 import org.apache.kafka.common.errors.SerializationException
 import org.apache.kafka.common.serialization.Serializer
 import org.slf4j.LoggerFactory
 import java.io.ByteArrayOutputStream
+import kotlin.reflect.KClass
+
+class AvroSerializer<T : SpecificRecordBase> : Serializer<T> {
+    private val encoders: HashMap<KClass<T>, BinaryMessageEncoder<T>> = HashMap()
 
-class AvroSerializer<T : SpecificRecord>(private val encoder: BinaryMessageEncoder<T>) : Serializer<T> {
     companion object {
         private val logger = LoggerFactory.getLogger(AvroSerializer::class.java)
     }
 
+    private fun getEncoder(message: T): BinaryMessageEncoder<T> {
+        var encoder = encoders[message::class]
+        if (encoder == null) {
+            encoder = BinaryMessageEncoder<T>(SpecificData(), message.schema)
```

Comment:
Je moet de encoder hier nog in de HashMap zetten.
En ik zou hier een nieuwe variabele voor gebruiken. Dan hoeft de encoder variabele niet mutable te zijn.